src/test/java/HandleTruthTest.java

```diff
@@ -1,8 +1,97 @@
+                        "\n" +
+                        "And it is no wonder that that people keep pointing out how the presidency has aged me. I look so old John Boehner’s already invited Benjamin Netanyahu to speak at my funeral.\n" +
+                        "\n" +
+                        "Meanwhile, Michelle hasn’t aged a day. I ask her what her secret is and she just says “fresh fruits and vegetables.” It’s aggravating.\n" +
+                        "\n" +
+                        "Fact is though, at this point my legacy is finally beginning to take shape. The economy is getting better. Nine in ten Americans now have health coverage. Today thanks to Obamacare you no longer have to worry about losing your insurance if you lose your job. You’re welcome, Senate democrats.\n " +
+                        "\n" +
+                        "Look, it is true I have not managed to make everybody happy. Six years into my presidency some people still say I’m arrogant, aloof, condescending. Some people are so dumb. No wonder I don’t meet with them. And that’s not all people say about me. A few weeks ago, Dick Cheney says he thinks I’m the worst president of his lifetime. Which is interesting because I think Dick Cheney is the worst president of my lifetime. Quite a coincidence. I mean everybody’s got something to say these days.\n" +
+                        "\n" +
+                        "Mike Huckabee recently said people shouldn’t join our military until a true conservative is elected president. Think about that. It was so outrageous 47 Ayatollahs wrote us a letter trying to explain to Huckabee how our system works.\n" +
+                        "\n" +
+                        "It gets worse. Just this week, Michele Bachmann actually predicted that I would bring about the biblical end of days. Now, that’s a legacy. That’s big. I mean, Lincoln, Washington, they didn’t do that.\n" +
+                        "\n"
+        };
+        for (int i = 0; i < msgs.length; i++) {
+            TreeMap<Integer, Set<String>> answer = HandleTruth.wordCount(msgs[i]);
+            if (msgs[i] == null) {
+                assertNull(answer);
+            } else {
+                Integer prev = Integer.MAX_VALUE;
+                for (Map.Entry<Integer, Set<String>> entry : answer.entrySet()) {
+                    assertTrue(entry.getKey() < prev);
+                    prev = entry.getKey();
+                    System.out.println(entry);
+                    assertFalse(entry.getValue().contains(" "));
+                    assertFalse(entry.getValue().contains("\\W"));
+                    if (entry.getValue().contains("'")) {
+                        int apostrophe = entry.getValue().toString().indexOf("'");
+                        Character singleChar = entry.getValue().toString().charAt(apostrophe + 1);
+                        assertFalse(singleChar.equals(' '));
```

Comment:
the test logic is a bit flawed here. During verification, we want to check the output is exactly as it should be instead of a specific logic. for example 

```java
assertTrue(entry.getKey() < prev);
```

this is validating the logic of a tree map being in the increasing order. the answer from wordCount could be wrong while still passing this. In fact, as long as the output map ordered with increasing keys, any output would pass this.

Another issue is the other part of the checks. these are mostly checking that the output should not have something instead of it should have something. 

Please see https://github.com/pdgetrf/you-cant-handle-the-truth/blob/solution/src/test/java/HandleTruthTest.java as an example of how we want to test this.